File: frontend/admin/packages/modules/system/src/api/system/admin/v1/base_session.ts
import service from "@liujitcn/kratos-admin-core/request";
import type {
  BaseSession,
  ListCurrentBaseSessionsRequest,
  ListCurrentBaseSessionsResponse,
  GetCurrentBaseSessionRequest,
  PageOnlineBaseSessionsRequest,
  PageOnlineBaseSessionsResponse,
  RevokeAllBaseSessionsRequest,
  RevokeBaseSessionRequest
} from "@liujitcn/kratos-admin-system/rpc/system/admin/v1/base_session";
import type { Empty } from "@liujitcn/kratos-admin-system/rpc/google/protobuf/empty";

const SESSION_URL = "/v1/admin/base/session";

/** BaseSessionService Admin会话管理服务。 */
export class BaseSessionServiceImpl {
  /** 查询当前用户全部有效会话。 */
  ListCurrentBaseSessions(request: ListCurrentBaseSessionsRequest): Promise<ListCurrentBaseSessionsResponse> {
    return service<ListCurrentBaseSessionsRequest, ListCurrentBaseSessionsResponse>({
      url: `${SESSION_URL}/current-user`,
      method: "get",
      params: request
    });
  }

  /** 查询当前用户会话。 */
  GetCurrentBaseSession(request: GetCurrentBaseSessionRequest): Promise<BaseSession> {
    return service<GetCurrentBaseSessionRequest, BaseSession>({
      url: `${SESSION_URL}/current`,
      method: "get",
      params: request
    });
  }

  /** 撤销当前用户全部会话。 */
  RevokeAllBaseSessions(request: RevokeAllBaseSessionsRequest): Promise<Empty> {
    return service<RevokeAllBaseSessionsRequest, Empty>({
      url: `${SESSION_URL}/revoke-all`,
      method: "put",
      data: request
    });
  }

  /** 分页查询当前在线用户会话。 */
  PageOnlineBaseSessions(request: PageOnlineBaseSessionsRequest): Promise<PageOnlineBaseSessionsResponse> {
    return service<PageOnlineBaseSessionsRequest, PageOnlineBaseSessionsResponse>({
      url: `${SESSION_URL}/online`,
      method: "get",
      params: request
    });
  }

  /** 下线指定用户会话。 */
  RevokeBaseSession(request: RevokeBaseSessionRequest): Promise<Empty> {
    return service<RevokeBaseSessionRequest, Empty>({
      url: `${SESSION_URL}/${request.session_id}`,
      method: "delete"
    });
  }
}

/** defBaseSessionService Admin会话管理服务实例。 */
export const defBaseSessionService = new BaseSessionServiceImpl();

File: backend/internal/service/system/admin/v1/base_session.go
package admin

import (
	"context"
	"fmt"

	"github.com/go-kratos/kratos/v3/log"
	adminv1 "github.com/liujitcn/kratos-admin/backend/api/gen/go/system/admin/v1"
	biz "github.com/liujitcn/kratos-admin/backend/internal/biz/system/admin"
	"github.com/liujitcn/kratos-core/errorsx"
	"google.golang.org/protobuf/types/known/emptypb"
)

// BaseSessionService 提供 Admin 会话管理接口。
type BaseSessionService struct {
	adminv1.UnimplementedBaseSessionServiceServer                      // 提供未实现 RPC 的默认返回。
	baseSessionCase                               *biz.BaseSessionCase // 负责当前会话查询和令牌撤销。
}

// NewBaseSessionService 创建 Admin 会话管理服务。
func NewBaseSessionService(baseSessionCase *biz.BaseSessionCase) *BaseSessionService {
	return &BaseSessionService{baseSessionCase: baseSessionCase}
}

// ListCurrentBaseSessions 查询当前用户全部有效会话。
func (s *BaseSessionService) ListCurrentBaseSessions(ctx context.Context, req *adminv1.ListCurrentBaseSessionsRequest) (*adminv1.ListCurrentBaseSessionsResponse, error) {
	res, err := s.baseSessionCase.ListCurrentBaseSessions(ctx)
	if err != nil {
		log.Error(fmt.Sprintf("ListCurrentBaseSessions %v", err))
		return nil, errorsx.WrapInternal(err, "查询本人会话失败")
	}
	return res, nil
}

// GetCurrentBaseSession 查询当前用户会话。
func (s *BaseSessionService) GetCurrentBaseSession(ctx context.Context, req *adminv1.GetCurrentBaseSessionRequest) (*adminv1.BaseSession, error) {
	res, err := s.baseSessionCase.GetCurrentBaseSession(ctx)
	if err != nil {
		log.Error(fmt.Sprintf("GetCurrentBaseSession %v", err))
		return nil, errorsx.WrapInternal(err, "查询当前会话失败")
	}
	return res, nil
}

// RevokeAllBaseSessions 撤销当前用户的全部会话。
func (s *BaseSessionService) RevokeAllBaseSessions(ctx context.Context, req *adminv1.RevokeAllBaseSessionsRequest) (*emptypb.Empty, error) {
	err := s.baseSessionCase.RevokeAllBaseSessions(ctx)
	if err != nil {
		log.Error(fmt.Sprintf("RevokeAllBaseSessions %v", err))
		return nil, errorsx.WrapInternal(err, "撤销全部会话失败")
	}
	return new(emptypb.Empty), nil
}

// PageOnlineBaseSessions 分页查询当前在线用户会话。
func (s *BaseSessionService) PageOnlineBaseSessions(ctx context.Context, req *adminv1.PageOnlineBaseSessionsRequest) (*adminv1.PageOnlineBaseSessionsResponse, error) {
	res, err := s.baseSessionCase.PageOnlineBaseSessions(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("PageOnlineBaseSessions %v", err))
		return nil, errorsx.WrapInternal(err, "查询在线用户失败")
	}
	return res, nil
}

// RevokeBaseSession 下线指定用户会话。
func (s *BaseSessionService) RevokeBaseSession(ctx context.Context, req *adminv1.RevokeBaseSessionRequest) (*emptypb.Empty, error) {
	err := s.baseSessionCase.RevokeBaseSession(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("RevokeBaseSession %v", err))
		return nil, errorsx.WrapInternal(err, "下线在线用户失败")
	}
	return new(emptypb.Empty), nil
}
